### PR TITLE
[MDETR] VQA fine-tuning

### PR DESCRIPTION
**Pull request by ebsmothers**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #200
* __->__ #199
* #198
* #181
* #180
* #179

Summary:
Add datamodule, optimizer, and main training loop for VQA fine-tuning.

Test plan:

```
CUBLAS_WORKSPACE_CONFIG=:4096:8 python -m torch.distributed.launch --nproc_per_node=4
--use_env vqa_finetune.py --dataset_config vqa.json --ema --epochs 125 --epoch_chunks 3
--lr_drop 150 --load /data/home/ebs/data/mdetr/checkpoints/pretrained_resnet101_checkpoint.pth
--qa_loss_coef 25 --batch_size 4 --lr 1.4e-4 --lr_backbone 1.4e-5 --text_encoder_lr 7e-5
```

Also running on 4 nodes via run_with_submitit, but will not commit this file as it is
pretty much a direct clone of the file in MDETR and torchvision repos. Will update with
results when the run finishes.

Differential Revision: [D38160124](https://our.internmc.facebook.com/intern/diff/D38160124)

### CODE REVIEW

File: examples/mdetr/optimizer.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Copyright (c) Aishwarya Kamath & Nicolas Carion. Licensed under the Apache License 2.0. All Rights Reserved
```
Comment (RdoubleA) [inline examples/mdetr/optimizer.py:7]: should we remove this line and add a comment with a link to the code ref?

**Review by RdoubleA — APPROVED** [review]

File: examples/mdetr/optimizer.py (hunk 1 above)
Comment (ebsmothers) [reply examples/mdetr/optimizer.py:7]: Yeah I plan to add a NOTICES file under the mdetr/examples root to cover everything in there

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38160124).

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38160124).

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38160124).

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38160124).

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38160124).

**Pull request closed without merging**